auto-dark.el

```diff
@@ -148,19 +146,67 @@ Argument APPEARANCE should be light or dark."
   (when (timerp auto-dark--timer)
     (cancel-timer auto-dark--timer)))
 
+(defun auto-dark--register-dbus-listener ()
+  "Ask D-Bus to send us a signal on theme change and add a callback
+to change the theme."
+  (setq auto-dark--dbus-listener-object
+        (dbus-register-signal
+         :session
+         "org.freedesktop.portal.Desktop"
+         "/org/freedesktop/portal/desktop"
+         "org.freedesktop.portal.Settings"
+         "SettingChanged"
+         (lambda (path var val)
+           (when (and (string= path "org.freedesktop.appearance")
+		      (string= var "color-scheme"))
+             (auto-dark--set-theme (pcase (car val) (0 'light) (1 'dark))))))))
+
+(defun auto-dark--unregister-dbus-listener ()
+  "Ask D-Bus to send us a signal on theme change and add a callback
+to change the theme."
+  (dbus-unregister-object auto-dark--dbus-listener-object))
+
+(defun auto-dark--register-change-listener ()
+  "Register a listener to listen for the system theme to change."
+  (cond
+   ((auto-dark--use-ns-system-appearance)
+    (add-hook 'ns-system-appearance-change-functions #'auto-dark--set-theme))
+   ((auto-dark--use-dbus)
+    (auto-dark--register-dbus-listener))
+   (t (auto-dark-start-timer))))
+
+(defun auto-dark--unregister-change-listener ()
+  "Remove an existing listener for the system theme."
+  (cond
+   ((auto-dark--use-ns-system-appearance)
+    (remove-hook 'ns-system-appearance-change-functions #'auto-dark--set-theme))
+   ((auto-dark--use-dbus)
+    (auto-dark--unregister-dbus-listener))
+   (t (auto-dark-stop-timer))))
+
+(defun auto-dark--use-ns-system-appearance ()
+  "Determine whether we should use the ns-system-appearance-* functions."
+  (boundp 'ns-system-appearance-change-functions))
+
+(defun auto-dark--use-dbus ()
+  "Determine whether we should use the dbus-* functions."
+  (and (fboundp 'dbus-register-signal)
+       (fboundp 'dbus-list-activatable-names)
+       (member "org.freedesktop.portal.Desktop"
+               (dbus-list-activatable-names :session))
+       t)) ;; t only here to make this a boolean
+
 ;;;###autoload
 (define-minor-mode auto-dark-mode
   "Toggle `auto-dark-mode' on or off."
   :group 'auto-dark
   :global t
   :lighter "AD"
   (if auto-dark-mode
-      (if (boundp 'ns-system-appearance-change-functions)
-          (add-hook 'ns-system-appearance-change-functions #'auto-dark--ns-set-theme)
-        (auto-dark-start-timer))
-    (if (boundp 'ns-system-appearance-change-functions)
-        (remove-hook 'ns-system-appearance-change-functions #'auto-dark--ns-set-theme)
-      (auto-dark-stop-timer))))
+      (progn
+	(auto-dark--check-and-set-dark-mode)
```

Comment:
Please format the codes.

Reply:
As far as I'm aware, there's no standard formatting for Emacs Lisp.  There are a few opinionated packages, e.g. [elisp-format](https://github.com/Yuki-Inoue/elisp-format) and [elfmt](https://github.com/riscy/elfmt), but they vary widely on what they consider to be "formatted."

Is it the tabs that you don't like?  I originally kept the tabs as Emacs inserted them thinking that this was the convention for this file, but I see now that originally they were only used in the Applescript in `auto-dark--is-dark-mode-builtin`.  I'd be happy to convert them to spaces if that's your preference.

Reply:
Personally I haven't found yet a config for elisp-format that pleases me :)

Originally I've used both spaces and tabs as `indent-tabs-mode` was "on" for elisp-mode and  I did not noticed that before.

That said, I tend to a `spaces only` approach in other languages. What do you think we should do here?